refinery/tool_manager/tests.py

```diff
@@ -595,6 +594,66 @@ def test_for_proper_parameters_in_response(self):
                       ToolDefinition.VISUALIZATION):
                     self.assertNotIn("galaxy_workflow_step", parameter.keys())
 
+    def test_request_from_owned_dataset_shows_all_tool_defs(self):
+        self.assertNotEqual(len(self.get_response.data), 0)
+        for tool_definition in self.get_response.data:
+            tool_definition = ToolDefinition.objects.get(
+                uuid=tool_definition["uuid"]
+            )
+            self.assertIn(tool_definition, ToolDefinition.objects.all())
+
+    def test_request_from_public_dataset_shows_vis_tools_only(self):
+        get_request = self.factory.get(
+            "{}?dataSetUuid={}".format(
+                self.tool_defs_url_root,
+                self.public_dataset.uuid
+            )
+        )
+        force_authenticate(get_request, self.user)
+        get_response = self.tool_defs_view(get_request)
+        self.assertNotEqual(len(get_response.data), 0)
+        for tool_definition in get_response.data:
+            tool_definition = ToolDefinition.objects.get(
+                uuid=tool_definition["uuid"]
+            )
+            self.assertIn(
+                tool_definition,
+                ToolDefinition.objects.filter(
+                    tool_type=ToolDefinition.VISUALIZATION
+                )
+            )
+
+    def test_no_query_params_in_get_yields_bad_request(self):
+        get_request = self.factory.get(self.tool_defs_url_root)
+        force_authenticate(get_request, self.user)
+        get_response = self.tool_defs_view(get_request)
+        self.assertEqual(get_response.status_code, 400)
+
+    def test_bad_query_params_in_get_yields_bad_request(self):
+        get_request = self.factory.get(
+            "{}?coffee={}".format(
+                self.tool_defs_url_root,
+                self.dataset.uuid
+            )
+        )
+        force_authenticate(get_request, self.user)
+        get_response = self.tool_defs_view(get_request)
+        self.assertEqual(get_response.status_code, 400)
+
+    def test_missing_dataset_in_get_yields_bad_request(self):
+        dataset_uuid = self.dataset.uuid
+        self.dataset.delete()
+
+        get_request = self.factory.get(
+            "{}?dataSetUuid={}".format(
+                self.tool_defs_url_root,
+                dataset_uuid
+            )
+        )
+        force_authenticate(get_request, self.user)
+        get_response = self.tool_defs_view(get_request)
+        self.assertEqual(get_response.status_code, 400)
```

Comment:
Is there anything in the bodies of the responses? Sometime a human-readable error message there could help with debugging, and it could be worthwhile to test against it.